core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -175,14 +192,43 @@
 async def _request_share_next_screen(ctx: wire.Context, mnemonic_type: int) -> None:
     if mnemonic_type == mnemonic.TYPE_SLIP39:
         remaining = storage.recovery.get_remaining()
+        shares_remaining = storage.recovery.get_slip39_shares_remaining()
         if not remaining:
             # 'remaining' should be stored at this point
             raise RuntimeError
-        if remaining == 1:
-            text = "1 more share"
+        if shares_remaining and len(shares_remaining.strip(b"\x00")) > 0:
```

Comment:
I believe the storage should return the data in such format, it is easy to use. See the comment above in storage.py.

Reply:
fixed with storage refactor